coil-base/src/main/java/coil/memory/HardwareBitmapService.kt

```diff
@@ -20,18 +22,30 @@ internal sealed class HardwareBitmapService {
         operator fun invoke() = when {
             SDK_INT < O || IS_BLACKLISTED -> ImmutableHardwareBitmapService(false)
             SDK_INT == O || SDK_INT == O_MR1 -> LimitedFileDescriptorHardwareBitmapService
-            else -> ImmutableHardwareBitmapService(true)
+            else -> ValidateHardwareBitmapService
         }
     }
 
     /** Return true if we can currently use [Bitmap.Config.HARDWARE]. */
-    abstract fun allowHardware(size: Size): Boolean
+    abstract fun allowHardware(size: Size, target: Target?): Boolean
 }
 
 /** Returns a fixed value for [allowHardware]. */
 private class ImmutableHardwareBitmapService(private val allowHardware: Boolean) : HardwareBitmapService() {
 
-    override fun allowHardware(size: Size) = allowHardware
+    override fun allowHardware(size: Size, target: Target?) = allowHardware
+}
+
+/* Validates that the view is ViewTarget and checks if its hardware accelerated */
+private object ValidateHardwareBitmapService : HardwareBitmapService() {
+
+    override fun allowHardware(size: Size, target: Target?): Boolean {
+        if (target != null && target is ViewTarget<*>) {
```

Comment:
Let's move this check to `Request.isConfigValidForHardware` and keep `ImmutableHardwareBitmapService` as is. `HardwareBitmapService` is supposed to handle device-specific and file system limitations for hardware bitmaps.